wordpress-mobile/WordPress-Android
# Feature/save post for later snackbar

Fixes #7720 

This PR adds a Snackbar that appears after you bookmark a post in post details footer, on in post list (excluding Saved Posts filter) and takes you to the Saved Posts lists.

There is a slight deviation from @iamthomasbishop's design.
On the post details screen, the snackbar appears over the bottom bar:
[![https://gyazo.com/77cfc57ae508c7b7c53a138a89b9d219](https://i.gyazo.com/77cfc57ae508c7b7c53a138a89b9d219.png)](https://gyazo.com/77cfc57ae508c7b7c53a138a89b9d219)

There are a couple of reasons for this:
Technical - it's surprisingly hard to make it appear not at the bottom of the screen! There is a way to do it, but it requires significant changes to the Post Details, that involve CoordinatorLayout and tying bottom bar to it (currently, bottom bar animates out when scrolling, and forcing Snackbar to do same is not fun).
Weird edge cases, like showing snack bar above the bottom bar opens users to a possibility of pressing "VIEW ALL" for a not bookmarked post (SAVE -> UNSAVE -> VIEW ALL while it's still on a screen).

PS: This PR assumes that there is only one Bookmark tag.

To test 1:

1. Oper Reader with not "Saved Posts" filter.
2. Press on a Bookmark button.
3. Observe Snackbar.
4. Press a View All button with all your might.
5. Notice that Saved Posts filter is selected and Bookmarked posts are in the list.

To test 2:
1. Open Post details.
2. Press "SAVE" button.
3. Observe Snackbar.
4. Press a View All button with all the remaining strength.
5. Notice that you were magically transferred to Reader at the Main activity, and Saved Posts filter is selected.

To test 3:
1. Navigate to Notification's tab.
2. Open some Post Details from there.
3. Press "SAVE" button.
4. Rest of the steps. You know the drill by this point.

To test 4:
1. Navigate to Notification's tab.
2. Open some Post Details from there.
3. Click on the Blog name to open that blog's Post list.
4. Click on the Bookmark icon in the list.
5. (Rest of the steps)
6. Profit!

To test 5:
1. Navigate to Reader and select Saved Posts filter.
2. Click on the Bookmark icon in the post.
3. Keep clicking.
4. Make sure not rogue Snackbars appear.
